### PR TITLE
Rns 106 add gas estimation to sdk

### PR DESCRIPTION
**Pull request by LordCheta**

### CODE REVIEW

File: src/PartnerRegistrar.ts [hunk 1]
```diff
@@ -84,18 +128,33 @@ export class PartnerRegistrar {
...
+      case 'register':
+        return this.registerOp(args[0], args[1] as string, args[2] as string, args[3] as BigNumber, args[4] as BigNumber, args[5] as string).estimateGas()
+
+      case 'renew':
+        return this.renewOp(args[0], args[1] as BigNumber, args[2] as BigNumber).estimateGas()
+
+      case 'commitAndRegister':
+        return this.commitAndRegisterOp(args[0], args[1] as string, args[2] as BigNumber, args[3] as BigNumber, args[4] as string, args[5] as string).estimateGas()
+      default:
+        return this.commitOp(args[0], args[1] as string, args[2] as BigNumber, args[3] as string).estimateGas()
```
Comment (OnyekaIjeh) [inline src/PartnerRegistrar.ts]: We should return an error for operations that are not supported

File: src/PartnerRegistrar.ts [hunk 2]
```diff
@@ -84,18 +128,33 @@ export class PartnerRegistrar {
...
     }
   }
 
+  private makeCommitment (label: string, owner: string, secret: string, duration: BigNumber, addr: string | undefined): string {
+    // return await this.partnerRegistrar.makeCommitment(hashLabel(label), owner, secret, duration, addr ?? owner)
```
Comment (OnyekaIjeh) [inline src/PartnerRegistrar.ts]: remove comment

File: src/PartnerRegistrar.ts [hunk 3]
```diff
@@ -113,7 +172,7 @@ export class PartnerRegistrar {
    * @param amount the amount for the name registration
    * @param addr the address to set for the name resolution
    */
-  async register (label: string, owner: string, secret: string, duration: BigNumber, amount: BigNumber, addr?: string): Promise<boolean> {
+  registerOp (...args: RegisterArgs): OperationResult<boolean> {
```
Comment (OnyekaIjeh) [inline src/PartnerRegistrar.ts]: the operator functions should be private and the arguments should be clearly stated

File: src/PartnerRegistrar.ts (hunk 1 above)
Comment (sergio-jara) [reply src/PartnerRegistrar.ts]: yes. I dont know how far typescript takes care of validation tho. I'm guessing it autogenerates the code that validates for the string values once the dist transpilation happens.

**Review by sergio-jara — APPROVED** [review]

**Pull request merged**